src/compactor_executor.rs

```diff
@@ -131,34 +131,29 @@ impl TokioCompactionExecutorInner {
             // filter out any expired entries -- eventually we can consider
             // abstracting this away into generic, pluggable compaction filters
             // but for now we do it inline
-            let kv = match raw_kv.attributes.expire_ts {
+            let kv = match raw_kv.expire_ts {
                 Some(expire_ts) if expire_ts <= now => {
                     // insert a tombstone instead of just filtering out the
                     // value in the iterator because this may otherwise "revive"
                     // an older version of the KV pair that has a larger TTL in
                     // a lower level of the LSM tree
-                    KeyValueDeletable {
+                    RowEntry {
                         key: raw_kv.key,
                         value: Tombstone,
-                        attributes: RowAttributes {
-                            ts: raw_kv.attributes.ts,
-                            expire_ts: None,
-                        },
+                        seq: raw_kv.seq,
+                        flags: raw_kv.flags,
```

Comment:
if we're putting the expire ts in the flags, we probably want to remove `has_expire_ts` from the flags in this situation, right? 

WDYT about haveing `RowEntry::new` compute the flags instead of passing in the flags based on whether the `expire_ts` and `create_ts` are `None`? That way we won't have to make sure they're always aligned

Reply:
> why do we need to store flags in RowEntry

thank you for the reminding,, i was planned to remove the `flags` in RowEntry but forgot to do it. all the flags so far could be derived from these optional flags, thus can view it as an encoding/decoding detail.